lib/services/livesync/playground/preview-app-livesync-service.ts

```diff
@@ -0,0 +1,136 @@
+		} catch (err) {
+			this.$logger.warn(`Unable to apply changes on device ${device.id}. Error is: ${JSON.stringify(err, null, 2)}.`);
+		}
+	}
+
+	private async syncFilesOnDevice(data: IPreviewAppLiveSyncData, device: Device, files?: string[]): Promise<void> {
+		const { appFilesUpdaterOptions, env, projectDir } = data;
+		const platform = device.platform;
+		const projectData = this.$projectDataService.getProjectData(projectDir);
+		const platformData = this.$platformsData.getPlatformData(platform, projectData);
+
+		await this.preparePlatform(platform, appFilesUpdaterOptions, env, projectData);
+
+		const payloads = this.getFilePayloads(platformData, projectData, files);
+		await this.$previewSdkService.applyChanges(payloads);
+	}
+
+	private getFilePayloads(platformData: IPlatformData, projectData: IProjectData, files?: string[]): FilePayload[] {
+		const appFolderPath = path.join(projectData.projectDir, APP_FOLDER_NAME);
+		const platformsAppFolderPath = path.join(platformData.appDestinationDirectoryPath, APP_FOLDER_NAME);
+
+		if (files) {
+			files = files.map(file => path.join(platformsAppFolderPath, path.relative(appFolderPath, file)));
+		} else {
+			files = this.$projectFilesManager.getProjectFiles(platformsAppFolderPath);
+		}
+
+		const filesToTransfer = files
+			.filter(file => file.indexOf(TNS_MODULES_FOLDER_NAME) === -1)
+			.filter(file => file.indexOf(APP_RESOURCES_FOLDER_NAME) === -1)
+			.filter(file => !_.includes(this.excludedFiles, path.basename(file)))
+			.filter(file => !_.includes(this.excludedFileExtensions, path.extname(file)));
+
+		this.$logger.trace(`Transferring ${filesToTransfer.join("\n")}.`);
+
+		const payloads = filesToTransfer
+			.map(file => {
+				const filePayload: FilePayload = {
+					event: PreviewSdkEventNames.CHANGE_EVENT_NAME,
+					file: path.relative(platformsAppFolderPath, file),
+					binary: isTextOrBinary.isBinarySync(file),
+					fileContents: ""
+				};
+
+				if (filePayload.binary) {
+					const bitmap =  <string>this.$fs.readFile(file);
+					const base64 = new Buffer(bitmap).toString('base64');
+					filePayload.fileContents = base64;
+				} else {
+					filePayload.fileContents = this.$fs.readText(file);
+				}
+
+				return filePayload;
+			});
+
+		return payloads;
+	}
+
+	private async preparePlatform(platform: string, appFilesUpdaterOptions: IAppFilesUpdaterOptions, env: Object, projectData: IProjectData): Promise<void> {
+		const nativePrepare = { skipNativePrepare: true };
+		const config = <IPlatformOptions>{};
+		const platformTemplate = <string>null;
+		const prepareInfo = {
+			platform,
+			appFilesUpdaterOptions,
+			env,
+			projectData,
+			nativePrepare,
+			config,
+			platformTemplate,
+			skipCopyTnsModules: true,
+			skipCopyAppResourcesFiles: true
+		};
+		await this.$platformService.preparePlatform(prepareInfo);
+	}
+
+	private showWarningsForNativeFiles(files: string[]): void {
+		if (files && files.length) {
+			for (const file of files) {
```

Comment:
Instead of using `if` and `for`, you can use `_.each` - it handles cases where even undefined is passed

Reply:
Or you can do:
```TypeScript
_.filter(files, file => file.indexOf(APP_RESOURCES_FOLDER_NAME) > -1)
   .forEach(file => this.$logger.warn(`Unable to apply changes from ${APP_RESOURCES_FOLDER_NAME} folder. You need to build your application in order to make changes in ${APP_RESOURCES_FOLDER_NAME} folder.`));
```